File: 1909.cpp
#include <vector>

using namespace std;

class Solution {
public:
    bool canBeIncreasing(vector<int>& nums) {
        int cnt = 0;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] <= nums[i-1]) {
                if (i > 1 && nums[i] <= nums[i - 2]) 
                    nums[i] = nums[i-1];
                cnt++;
                if (cnt > 1) return false;
            }
        }
        return true;
    }
};

File: 1909.go
package leetcode

func canBeIncreasing(nums []int) bool {
	cnt := 0
	for i := 1; i < len(nums); i++ {
		if (nums[i] <= nums[i-1]) {
			if (i >= 2 && nums[i] <= nums[i-1]) {
				nums[i] = nums[i-1]
			}
			cnt++
			if (cnt > 1) {
				return false
			}
		}
	}
	return true;
}